addon/classes/Table.js

```diff
@@ -139,42 +139,46 @@ export default class Table extends Ember.Object.extend({
    * Replace all the row's content with content of the argument. If argument is an empty array rows will be cleared.
    * @method setRows
    * @param  {Array} rows
+   * @param  {Object} options
    * @return {Array} rows
    */
-  setRows(rows = []) {
-    return this.rows.setObjects(Table.createRows(rows));
+  setRows(rows = [], options = {}) {
+    return this.rows.setObjects(Table.createRows(rows, options));
   }
 
   /**
    * Push the object onto the end of the row array if it is not already present.
    * @method addRow
    * @param  {Object} row
+   * @param  {Object} options
    */
-  addRow(row) {
+  addRow(row, options = {}) {
     if (row instanceof Row) {
       this.rows.addObject(row);
     } else if (isNone(this.rows.findBy('content', row))) {
-      this.pushRow(row);
+      this.pushRow(row, options);
     }
   }
 
   /**
    * Push the objects onto the end of the row array if it is not already present.
    * @method addRows
    * @param  {Array} rows
+   * @param  {Object} options
    */
-  addRows(rows = []) {
-    rows.forEach(r => this.addRow(r));
+  addRows(rows = [], options = {}) {
```

Comment:
@curtisgr I totally understand your use-case. You have 2 options. 

1. In your while loop, just do 

```js
this.table.addRow({
    date: date.format('YYYY-MM-DD'),
    notes: holiday ? holiday.name : ''
  }, {
    classNames: classNames
  });
```

2. Instead of using `new Row` use the static method `Table.createRow{{}, {})` which is public API. 

In conclusion, I think either solution would work and will be using public APIs. Do you have some other proposition?